polygon.js

```diff
@@ -0,0 +1,216 @@
+                    item.x[j] += evt.clientX - mouse.x;
+                    item.y[j] += evt.clientY - mouse.y;
+                }
+                mouse.x = evt.clientX;
+                mouse.y = evt.clientY;
+                colorDraw(polygons, canvas, ctx);
+            }
+        });
+    });
+    canvas.addEventListener('mouseup', function(evt){
+        mouse.x = 0;
+        mouse.y = 0;
+        polygons.forEach(function(item, i, arr){
+            if(item.selected){
+                for(let j = 0; j < arr.length; j++){
+                    if(i == j) continue
+                    if(intOfPol(item, arr[j]) || overlappingOfPolygons(item, arr[j]) || overlappingOfPolygons(arr[j], item)){
+                        if(item.intersected.indexOf(j) == -1){
+                            item.intersected.push(j);
+                            arr[j].intersected.push(i);
+                        }
+                    } else if(item.intersected.indexOf(j) >= 0){
+                        arr[j].intersected.splice(arr[j].intersected.indexOf(i), 1);
+                        item.intersected.splice(item.intersected.indexOf(j), 1);
+                    }
+                } 
+            }
+            flag = 0;
+            item.selected = false;
+        });
+        colorDraw(polygons, canvas, ctx)
+    });
+}
+
+function initPolygon(n, m) {
+    polygons = [];
+    firstAngle = -PI / 2;
+    centerX = 50;
+    centerY = 50;
+    for (let i = 0; i < n; i++) {
+        let radius = getRandomArbitrary(25, 40),
+            numberOFAngles = getRandomArbitrary(3, m),
+            x = [];
+            y = [];
+
+        if(i) centerY += length; 
+        for (let j = 0; j < numberOFAngles; j++) {
+            x[j] = Math.round(centerX + radius * Math.cos(firstAngle + (2 * PI * j)/(numberOFAngles)));
+            y[j] = Math.round(centerY + radius * Math.sin(firstAngle + (2 * PI * j)/(numberOFAngles)));
+        }
+        polygons[i] = {
+            radius: radius,
+            x: x,
+            y: y,
+            intersected: [],
+            selected: false
+        }
+    }
+    return polygons;
+}
+
+function getRandomArbitrary(min, max) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+function colorDraw(polygons, canvas, ctx) {
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+    polygons.forEach(function(item, i, arr) {
+        ctx.fillStyle = "#000000"
+        if(item.intersected.length) {
+            ctx.fillStyle = "#FF0000"
+        };
+        ctx.beginPath();
+        ctx.moveTo(item.x[0], item.y[0]);
+        for(let i = 1; i < item.x.length; i++) {
+            ctx.lineTo(item.x[i], item.y[i]);
+        }
+        ctx.fill();
+    });
+}
+
+function inPolygon(x, y , polygon) {
+    npol = polygon.x.length;
+    j = npol - 1;
+    var c = false;
+    for (i = 0; i < npol;i++){
+        if ((((polygon.y[i] <= y) && (y < polygon.y[j])) || ((polygon.y[j] <= y) && (y < polygon.y[i]))) &&
```

Comment:
is it easy to understand this if condition? move to some method or at least named args